engine-mr/src/main/java/org/apache/kylin/engine/mr/common/AbstractHadoopJob.java

```diff
@@ -494,6 +494,17 @@ public static KylinConfig loadKylinConfigFromHdfs(SerializableConfiguration conf
     }
 
     public static KylinConfig loadKylinConfigFromHdfs(String uri) {
+        KylinConfig config = loadKylinConfigFromHdfsNotThreadLocal(uri);
+        
+        // This is a bad example where the thread local KylinConfig cannot be auto-closed due to 
+        // limitation of MR API. It works because MR task runs its own process. Do not copy.
+        @SuppressWarnings("unused")
+        SetAndUnsetThreadLocalConfig shouldAutoClose = KylinConfig.setAndUnsetThreadLocalConfig(config);
+
+        return config;
+    }
+
+    public static KylinConfig loadKylinConfigFromHdfsNotThreadLocal(String uri) {
```

Comment:
The method name is confusing. I think the root cause is the bad sample in AbstractHadoopJob.loadKylinConfigFromHdfs(), we should fix that.